freckle-app/tests/Freckle/App/TextualEnumSpec.hs

```diff
@@ -0,0 +1,95 @@
+module Freckle.App.TextualEnumSpec
+  ( spec
+  ) where
+
+import Freckle.App.Prelude
+
+import Data.Aeson qualified as JSON
+import Data.Csv qualified as CSV
+import Data.List.NonEmpty (NonEmpty ((:|)))
+import Database.Persist.Sql qualified as Sql
+import Freckle.App.Test.Properties.JSON
+import Freckle.App.Test.Properties.PathPiece
+import Freckle.App.Test.Properties.PersistValue
+import Freckle.App.TextualEnum
+import Servant qualified
+import Test.Hspec
+import Test.QuickCheck
+import Web.PathPieces qualified as Path
+
+data PrimaryColor
+  = Red
+  | Blue
+  | Yellow
+  deriving stock (Bounded, Enum, Eq, Show)
+  deriving
+    ( CSV.ToField
+    , CSV.FromField
+    , JSON.ToJSON
+    , JSON.FromJSON
+    , Servant.FromHttpApiData
+    , Servant.ToHttpApiData
+    , Path.PathPiece
+    , Sql.PersistField
+    , Arbitrary
+    )
+    via TextualEnum PrimaryColor
+
+instance EnumValue PrimaryColor where
+  toText = \case
```

Comment:
What about an example showing using https://hackage.haskell.org/package/base-4.20.0.1/docs/Data-Data.html#v:toConstr which would help DRY/standardize/make one less spot to forget to update when adding an enum value

Reply:
I'm :-1: in general on things that couple strings to Haskell identifiers, because it makes the code less robust against refactoring, leads to compromising what strings should be to deal with Haskell restrictions (e.g. wanting an enum string to be "type"), and makes it harder to search the code when you're expecting to be able to find a string literal appearing somewhere but it's actually derived from a Haskell identifier instead -- But sometimes I guess it's appropriate, and TIL about `toConstr`, so wouldn't argue against more docs.

Reply:
🤷 I get that pov. I do go back and forth. so much is automatically generated, and depending upon tradeoffs, it becomes unclear when specifically someone would be searching for a string vs somewhere else. But yea if you do autogen these, it really important that someone be able to decouple these when its needed, and sometimes stuff don't have "i need to roll my own" docs. 

either way perhaps its worth documenting/commenting/illustrating using it? i'm good with either way.